src/main/java/com/commercetools/sync/commons/utils/CustomTypeReferenceReplacementUtils.java

```diff
@@ -0,0 +1,41 @@
+package com.commercetools.sync.commons.utils;
+
+import io.sphere.sdk.types.Custom;
+import io.sphere.sdk.types.CustomFields;
+import io.sphere.sdk.types.CustomFieldsDraft;
+import io.sphere.sdk.types.CustomFieldsDraftBuilder;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class CustomTypeReferenceReplacementUtils {
+    /**
+     * Given a resource of type {@code T} that extends {@link Custom} (i.e. it has {@link CustomFields}, this method
+     * checks if the custom fields are existing (not null) and they are reference expanded. If they are then
+     * it returns a {@link CustomFieldsDraft} instance with the custom type key in place of the id of the reference.
+     * Otherwise, if its not reference expanded it returns a {@link CustomFieldsDraft} with the id not replaced. If the
+     * resource has no or null {@link Custom}, then it returns {@code null}.
```

Comment:
If a resource doesn't have a `Custom` it means the Custom is `null`. So maybe:
`If the resource has no or null {@link Custom}` -> `If {@link Custom} of the resource is null`?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/297/commits/7dcb9c6314c51346c2a59bb73feb70f44043e8b4